lib/fluent/plugin/buffer.rb

```diff
@@ -761,6 +771,15 @@ def write_step_by_step(metadata, data, format, splits_count, &block)
             begin
               while writing_splits_index < splits.size
                 split = splits[writing_splits_index]
+                if split.size == 1 && original_bytesize == 0
+                  big_record_size = format ? format.call(split).bytesize : split.first.bytesize
```

Comment:
The result of `format.call(split)` should be stored to pass it to following `chunk.concat`.
Otherwise `format.call(split)` will be called twice on non-error case, it's wasteful.

Reply:
Got it, I've fixed it.